Renovate does not update buildscript-gradle.lockfile

### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

_No response_

### If you're self-hosting Renovate, select which platform you are using.

None

### If you're self-hosting Renovate, tell us what version of the platform you run.

_No response_

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

When I use Renovate in a repository with [`buildscript-gradle.lockfile`](https://docs.gradle.org/current/userguide/dependency_locking.html#enabling_locking_on_configurations), Renovate is not able to update an existing `buildscript-gradle.lockfile`, see [minimal reproductions](https://github.com/chhch/renovate-gradle-lock-file-example).

### Relevant debug logs

<details><summary>Logs</summary>

```
Command failed: docker run --rm --name=renovate_sidecar --label=renovate_child -v "/mnt/renovate/gh/chhch/renovate-gradle-lock-file-example":"/mnt/renovate/gh/chhch/renovate-gradle-lock-file-example" -v "/tmp/renovate-cache":"/tmp/renovate-cache" -v "/tmp/containerbase":"/tmp/containerbase" -e GRADLE_OPTS -e BUILDPACK_CACHE_DIR -e CONTAINERBASE_CACHE_DIR -w "/mnt/renovate/gh/chhch/renovate-gradle-lock-file-example" docker.io/renovate/sidecar bash -l -c "install-tool java 16.0.2+7 && ./gradlew --console=plain -q properties"
openjdk version "16.0.2" 2021-07-20
OpenJDK Runtime Environment Temurin-16.0.2+7 (build 16.0.2+7)
OpenJDK 64-Bit Server VM Temurin-16.0.2+7 (build 16.0.2+7, mixed mode, sharing)

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'renovate-gradle-lock-file-example'.
> Could not resolve all files for configuration ':classpath'.
   > Could not resolve io.spring.dependency-management:io.spring.dependency-management.gradle.plugin:1.1.0.
     Required by:
         project :
      > Cannot find a version of 'io.spring.dependency-management:io.spring.dependency-management.gradle.plugin' that satisfies the version constraints:
           Dependency path ':renovate-gradle-lock-file-example:unspecified' --> 'io.spring.dependency-management:io.spring.dependency-management.gradle.plugin:1.1.0'
           Constraint path ':renovate-gradle-lock-file-example:unspecified' --> 'io.spring.dependency-management:io.spring.dependency-management.gradle.plugin:{strictly 1.0.15.RELEASE}' because of the following reason: dependency was locked to version '1.0.15.RELEASE'

   > Could not resolve io.spring.dependency-management:io.spring.dependency-management.gradle.plugin:{strictly 1.0.15.RELEASE}.
     Required by:
         project :
      > Cannot find a version of 'io.spring.dependency-management:io.spring.dependency-management.gradle.plugin' that satisfies the version constraints:
           Dependency path ':renovate-gradle-lock-file-example:unspecified' --> 'io.spring.dependency-management:io.spring.dependency-management.gradle.plugin:1.1.0'
           Constraint path ':renovate-gradle-lock-file-example:unspecified' --> 'io.spring.dependency-management:io.spring.dependency-management.gradle.plugin:{strictly 1.0.15.RELEASE}' because of the following reason: dependency was locked to version '1.0.15.RELEASE'


* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 41s
```

</details>


### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

--- a/lib/modules/manager/gradle/artifacts.ts
+++ b/lib/modules/manager/gradle/artifacts.ts
@@ -124,8 +124,6 @@ export async function updateArtifacts({
 
   try {
     const oldLockFileContentMap = await getFiles(lockFiles);
-
-    await writeLocalFile(packageFileName, newPackageFileContent);
     await prepareGradleCommand(gradlewFile);
 
     let cmd = `${gradlewName} --console=plain -q`;
@@ -159,6 +157,7 @@ export async function updateArtifacts({
       cmd += ` --update-locks ${updatedDepNames.map(quote).join(',')}`;
     }
 
+    await writeLocalFile(packageFileName, newPackageFileContent);
     await exec(cmd, execOptions);
 
     const res = await getUpdatedLockfiles(oldLockFileContentMap);

